torch/amp/grad_scaling.py

```diff
@@ -0,0 +1,382 @@
+            the CPU.
+            Another case where it makes sense to supply ``skip_if_inf=False`` is batch replay, where you replay this
+            iteration until inf/nan-free gradients are created.  By the time you reach the
+            :func:`optimizer.unscale_and_step()` call, you're sure the gradients do not contain inf/nan, so it's safe
+            to elide the check.  For other (typical) use, the default ``skip_if_inf=True`` is strongly recommended.
+        found_inf (torch.cuda.FloatTensor, optional):  If you've already created a 1-element Tensor that contains > 0.0
+            if the optimizer's current gradients have an inf/nan and 0.0 if they don't (by, for example, an earlier
+            call to ``optimizer.check_inf()``) you may pass this as ``found_inf``.  By default, you don't need to pass
+            ``found_inf`` manually.  :func:`unscale_and_step` will inspect the gradients for you, create a
+            ``found_inf`` tensor, and return it.  Leaving ``found_inf=None`` and letting :func:`unscale_and_step`
+            create it for you is typical.
+        scaling_enabled (bool, optional):  If ``False``, :func:`unscale_and_step` simply calls ``self.step()`` and
+            returns ``opt_ret, None, None``, where opt_ret is the default return value of ``self.step()``.
+
+    Returns:
+        A tuple ``opt_ret, found_inf, recommended_scale``.  ``found_inf`` is a ``torch.cuda.FloatTensor`` that
+        contains > 0 if an inf/nan was found during unscaling, and 0.0 otherwise.  ``recommended_scale`` is a
+        ``torch.cuda.FloatTensor`` containing the recommended gradient scale to use next iteration.  ``opt_ret``
+        is the default return value of the underlying ``self.step()`` call.  If ``scaling_enabled=False``, the return
+        values will be ``opt_ret, None, None``.
+    """
+    if not scaling_enabled:
+        return self.step(closure), None, None
+
+    if closure is not None:
+        raise ValueError("Closure use is not currently supported.  It's tricky, but not impossible, and we're trying "
+                         "to decide if it's worth implementing.  If you require closure use, please comment on "
+                         "https://github.com/pytorch/pytorch/issues/25081, which will help us gauge demand.")
+
+    found, recommended_scale = self.unscale(current_scale,
+                                            scale_growth_factor=scale_growth_factor,
+                                            scale_backoff_factor=scale_backoff_factor,
+                                            scale_scheduler=scale_scheduler)
+
+    if found_inf is None:
+        found_inf = found
+
+    if skip_if_inf:
+        if found_inf.item():
+            return None, found_inf, recommended_scale
+        else:
+            return self.step(), found_inf, recommended_scale
+    else:
+        return self.step(), found_inf, recommended_scale
```

Comment:
Since step is define by the class you're "decorating" I think you want to pass *args and **kwargs to it.